statement.go

```diff
@@ -46,6 +46,7 @@ type Statement struct {
 	attrs                []interface{}
 	assigns              []interface{}
 	scopes               []func(*DB) *DB
+	queryTypes           []bool
```

Comment:
Can we store it in `Settings` ? I haven't tested it, but could something wrong with this ?
```go
tx := db.Scope(...)
go func() {
	tx.Scan(...)
}()
go func() {
	tx.Scan(...)
}()
```

Reply:
It has the same problem unless we use thread-safe slice/array

Reply:
> It has the same problem unless we use thread-safe slice/array

Or can we use lock?